t/unit/app/test_exceptions.py

```diff
@@ -12,19 +12,12 @@ def test_when_datetime(self):
         assert x.humanize()
 
     def test_pickleable(self):
-        x = Retry('foo', KeyError(), when=datetime.utcnow())
-        y = pickle.loads(pickle.dumps(x))
-        assert repr(x) == repr(y)
-
-    def test_wrap_unpickleable_exception(self):
```

Comment:
I'm not comfortable removing this test.

Reply:
This test was added in previous commit in same PR.

Adding `celery.utils.serialization` to `celery.exceptions` leads to circural import error, so this test will demonstrate some "weakness", but don't propose good solution for a related problem.

Reply:
BTW, "Recursion Exceeded" is caused by `Retry.__repr__`, not by changes introduced in PR.

Reply:
> 
> 
> This test was added in previous commit in same PR.
> 

Oh now everything makes sense.